rskj-core/src/main/java/co/rsk/util/EthSwapUtil.java

```diff
@@ -0,0 +1,209 @@
+                timeLock
+        );
+
+        return HashUtil.keccak256(argumentsBytes);
+    }
+
+    public static byte[] encodePacked(Object...arguments) {
+        byte[] encodedArguments = new byte[]{};
+
+        for(Object arg: arguments) {
+            byte[] encodedArg = new byte[]{};
+            if(arg instanceof byte[]) {
+                SolidityType bytes32Type = SolidityType.getType(SolidityType.BYTES32);
+                encodedArg = bytes32Type.encode(arg);
+            } else if(arg instanceof RskAddress) {
+                SolidityType addressType = SolidityType.getType(SolidityType.ADDRESS);
+                byte[] encodedAddress = addressType.encode(((RskAddress) arg).toHexString());
+                encodedArg = org.bouncycastle.util.Arrays.copyOfRange(encodedAddress, 12, encodedAddress.length);
+            } else if(arg instanceof BigInteger) {
+                SolidityType uint256Type = SolidityType.getType(SolidityType.UINT);
+                encodedArg = uint256Type.encode(arg);
+            }
+
+            encodedArguments = ByteUtil.merge(encodedArguments, encodedArg);
+        }
+
+        return encodedArguments;
+    }
+
+    public static CallTransaction.Invocation getTxInvocation(Transaction newTx) {
+        String abi = "[{\"constant\":false,\"inputs\":[{\"name\":\"preimage\",\"type\":\"bytes32\"}, {\"name\":\"amount\",\"type\":\"uint256\"}, {\"name\":\"address\",\"type\":\"address\"}, {\"name\":\"timelock\",\"type\":\"uint256\"}],\"name\":\"claim\",\"outputs\":[],\"payable\":false,\"type\":\"function\"}]";
+        CallTransaction.Contract contract = new CallTransaction.Contract(abi);
+        return contract.parseInvocation(newTx.getData());
+    }
+
+    public static boolean isClaimTx(Transaction tx, Constants constants) {
+        return tx.getReceiveAddress() != null
+                && !tx.getReceiveAddress().toHexString().isEmpty()
+                && tx.getData() != null
+                && tx.getReceiveAddress().toHexString().equalsIgnoreCase(constants.getEtherSwapContractAddress())
+                && Arrays.equals(Arrays.copyOfRange(tx.getData(), 0, 4), Constants.CLAIM_FUNCTION_SIGNATURE);
+    }
+
+    public static boolean hasLockedFunds(Transaction tx, SignatureCache signatureCache, RepositorySnapshot repositorySnapshot) {
+        String swapHash = HexUtils.toUnformattedJsonHex(calculateSwapHash(tx, signatureCache));
+        byte[] key = HashUtil.keccak256(HexUtils.decode(HexUtils.stringToByteArray(swapHash + SWAPS_MAP_POSITION)));
+
+        DataWord swapRecord = repositorySnapshot.getStorageValue(tx.getReceiveAddress(), DataWord.valueOf(key));
+
+        return swapRecord != null;
+    }
+
+    private static Coin getLockedAmount(Transaction tx) {
+        CallTransaction.Invocation invocation = getTxInvocation(tx);
+        BigInteger amount = (BigInteger) invocation.args[1];
+
+        return Coin.valueOf(amount.longValue());
+    }
+
+    private static Coin getTxCost(Transaction tx,
```

Comment:
By this name the fuction returns the _transaction cost_ but then we have a method `tx.transactionCost(constants, activation, signatureCache)`
What is the difference between these two costs?

Reply:
I though I needed the if check this method is used in to get the correct tx cost value but it looks like I don't. I've removed the check so this method is not necessary here.